float32 entrada
float64 time
